config/confighttp/client_middleware_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package confighttp
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configmiddleware"
+	"go.opentelemetry.io/collector/extension/extensionmiddleware"
+)
+
+// testClientMiddleware is a test middleware that appends a string to the response body
+type testClientMiddleware struct {
+	name string
+}
+
+func (*testClientMiddleware) Start(_ context.Context, _ component.Host) error {
+	return nil
+}
+
+func (*testClientMiddleware) Shutdown(_ context.Context) error {
+	return nil
+}
```

Comment:
same comment about embedding the component funcs.

Reply:
Fixed. Note that I added a new test helper HTTPClientFunc which is the equivalent of net/http's HTTPHandlerFunc.